gatk-variantcalling.wdl

```diff
@@ -0,0 +1,134 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import "tasks/biopet/biopet.wdl" as biopet
+import "tasks/common.wdl" as common
+import "tasks/gatk.wdl" as gatk
+import "tasks/picard.wdl" as picard
+import "tasks/samtools.wdl" as samtools
+import "gvcf.wdl" as gvcf
+
+workflow GatkVariantCalling {
+    input {
+        Array[IndexedBamFile] bamFiles
+        String outputDir = "."
+        String vcfBasename = "multisample"
+        File referenceFasta
+        File referenceFastaDict
+        File referenceFastaFai
+        File dbsnpVCF
+        File dbsnpVCFIndex
+
+        File? regions
+        # scatterSize is on number of bases. The human genome has 3 000 000 000 bases.
+        # 1 billion gives approximately 3 scatters per sample.
+        Int scatterSize = 1000000000
+        Map[String, String] dockerImages = {
+          "picard":"quay.io/biocontainers/picard:2.20.5--0",
+          "gatk4":"quay.io/biocontainers/gatk4:4.1.0.0--0",
+          "biopet-scatterregions":"quay.io/biocontainers/biopet-scatterregions:0.2--0",
+        }
+    }
+
+    call biopet.ScatterRegions as scatterList {
+        input:
+            referenceFasta = referenceFasta,
+            referenceFastaDict = referenceFastaDict,
+            scatterSize = scatterSize,
+            regions = regions,
+            dockerImage = dockerImages["biopet-scatterregions"]
+    }
+
+    # Glob messes with order of scatters (10 comes before 1), which causes problems at gatherGvcfs
+    call biopet.ReorderGlobbedScatters as orderedScatters {
+        input:
+            scatters = scatterList.scatters,
+            # Dockertag not relevant here. Python script always runs in the same
+            # python container.
+    }
+
+    scatter (bam in bamFiles) {
+        # Call separate pipeline to allow scatter in scatter.
+        # Also this is needed. If there are 50 bam files, we need more scattering than
+        # when we have 1 bam file.
+        call gvcf.Gvcf as Gvcf {
+            input:
+                bam=bam.file,
+                bamIndex=bam.index,
+                scatterList=orderedScatters.reorderedScatters,
+                referenceFasta=referenceFasta,
+                referenceFastaDict=referenceFastaDict,
+                referenceFastaFai=referenceFastaFai,
+                dbsnpVCF=dbsnpVCF,
+                dbsnpVCFIndex=dbsnpVCFIndex,
+                outputDir=outputDir + "/samples/" + basename(bam.file, ".bam"),
+                dockerImages=dockerImages
```

Comment:
[Please add spaces around the `=`.](https://biowdl.github.io/styleGuidelines.html#expressions)